### DISCUSSION TITLE
Clarification about contributing to examples

### DISCUSSION THREAD

**Discussion by paravatha** [argoproj-labs/hera#896] [Q&A] [answered]
Hey Hera team,
I am trying to create a new ml-pipeline example [under use-case folder](https://github.com/argoproj-labs/hera/tree/main/examples/workflows/use-cases) following this section 
https://github.com/argoproj-labs/hera/blob/main/CONTRIBUTING.md#upstream-hera-examples

I don't understand this part

> Define your new workflow. Make sure that the target workflow you wish to export and test against is named w

is there name convention I need to follow to generate a yaml file out of my .py file?
Here is my repo https://github.com/paravatha/argo-workflows-hera/tree/main/examples/workflows/use-cases
I created a copy of `spark.py` called `spark-copy.py` which generates the corresponding yaml.
for my new example file [examples/workflows/use-cases/ml_infer_pipeline_spacy.py](https://github.com/paravatha/argo-workflows-hera/blob/main/examples/workflows/use-cases/ml_infer_pipeline_spacy.py) it does generate the yaml file

I am using Python3.11.6
Here are the commands I ran 
```
poetry install
make format
make test
make regenerate-test-data
```

**Comment by elliotgunton** [comment, marked as answer]
Naming the variable `w` [as you have done here](https://github.com/paravatha/argo-workflows-hera/blob/130e9c765f01e79a9f267be46ff4f0a52f3c6a59/examples/workflows/use-cases/ml_infer_pipeline_spacy.py#L112) is all that's needed! We can rephrase that part of the doc to clarify. Thanks!

**Comment by paravatha** [comment]
Thanks @elliotgunton 

Seems like my issues are
1. using f-string in my print statements https://github.com/argoproj-labs/hera/issues/584 
2. I am using using python 3.11, so `make test` defaults to that version
3. Then `regenerate-test-data` uses python3.8 by default
